### PR TITLE
Feat/check disk only label

### PR DESCRIPTION
**Pull request by absolutejam**
This allows the plugin to (o)nly check specific mount points.

## Pull Request Checklist

This is not in reference to an existing issue, but a feature I needed. This lets me create specific checks for certain, high priority disks.

#### General

- [x] Update Changelog following the conventions laid out on [Our CHANGELOG Guidelines ](https://github.com/sensu-plugins/community/blob/master/HOW_WE_CHANGELOG.md)

- [x] Update README with any necessary configuration snippets

- [ ] Binstubs are created if needed

- [x] RuboCop passes

- [x] Existing tests pass

### CODE REVIEW

File: bin/check-windows-disk.rb [hunk 1]
```diff
@@ -31,6 +31,10 @@ class CheckDisk < Sensu::Plugin::Check::CLI
   option :fstype,
          short: '-t TYPE',
          proc: proc { |a| a.split(',') }
+ 
+  option :onlymnt,
```
Comment (majormoses) [inline bin/check-windows-disk.rb]: I think a better name would be `mount_points`

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,8 @@ This CHANGELOG follows the format listed at [Our CHANGELOG Guidelines ](https://
 Which is based on [Keep A Changelog](http://keepachangelog.com/)
 
 ## [Unreleased]
+### Added
+- `check-windows-service.rb`: adds -o option to explicitly specify which drive mounts to check (@absolutejam)
```
Comment (majormoses) [inline CHANGELOG.md]: see comment regarding option name.

File: bin/check-windows-disk.rb [hunk 3]
```diff
@@ -31,6 +31,10 @@ class CheckDisk < Sensu::Plugin::Check::CLI
...
+         short: '-o MNT',
```
Comment (majormoses) [inline bin/check-windows-disk.rb]: if any short it should be `-m`

File: bin/check-windows-disk.rb [hunk 4]
```diff
@@ -69,6 +73,7 @@ def read_wmic
         next if _avail.nil?
         next if line.include?('System Reserved')
         next if line.include?('\Volume')
+        next if config[:onlymnt] && !config[:onlymnt].include?(mnt)
```
Comment (majormoses) [inline bin/check-windows-disk.rb]: see comment about name.

**Comment by majormoses** [comment]
Can I also get a [testing artifact](https://github.com/sensu-plugins/community/blob/master/PULL_REQUEST_PROCESS.md#7-testing-artifacts)?

File: bin/check-windows-disk.rb [hunk 5]
```diff
@@ -31,6 +31,10 @@ class CheckDisk < Sensu::Plugin::Check::CLI
   option :fstype,
          short: '-t TYPE',
          proc: proc { |a| a.split(',') }
+
```
Comment (majormoses) [inline bin/check-windows-disk.rb]: trailing whitespace.

**Comment by absolutejam** [comment]
Are there any example plugins that have tests? Ruby and rspec especially is alien to me, so I'm not doing a great job at making my test work, nevermind pass 🤦‍♂️

**Comment by majormoses** [comment]
check out any of the ones checked off here: https://github.com/sensu-plugins/community/issues/41

dcos was the initial one and probably has the most coverage.

**Comment by absolutejam** [comment]
@majormoses Changes taken into consideration.

Hope you don't mind the `.rubocop.yml` tweaks too, but they didn't make sense for rspec files.

File: CHANGELOG.md (hunk 2 above)
Comment (absolutejam) [reply CHANGELOG.md]: Changed

File: bin/check-windows-disk.rb (hunk 1 above)
Comment (absolutejam) [reply bin/check-windows-disk.rb]: Changed

File: bin/check-windows-disk.rb (hunk 3 above)
Comment (absolutejam) [reply bin/check-windows-disk.rb]: Changed

File: bin/check-windows-disk.rb (hunk 4 above)
Comment (absolutejam) [reply bin/check-windows-disk.rb]: Changed

File: bin/check-windows-disk.rb (hunk 5 above)
Comment (absolutejam) [reply bin/check-windows-disk.rb]: Fixed

File: bin/check-windows-disk.rb [hunk 6]
```diff
@@ -32,6 +32,11 @@ class CheckDisk < Sensu::Plugin::Check::CLI
          short: '-t TYPE',
          proc: proc { |a| a.split(',') }
 
+  option :mount_points,
+         short: '-o MNT',
```
Comment (majormoses) [inline bin/check-windows-disk.rb]: I think the short should be changed to `-m`.

**Comment by absolutejam** [comment]
Oops, thanks for spotting that.

Added tests to check both long and short flags, tweaking original tests to better emulate CLI args (Using `parse_options` instead of directly setting the values) 👍

**Comment by absolutejam** [comment]
Do you want the commits squashing for tidiness?

**Comment by majormoses** [comment]
Before githb offered it I often requested people to. Now I generally use the squash and rebase option to accept a contribution when there are a lot of commits with messages with little overall value. If the commits paint a story of how we got there and it is valuable I do a rebase and merge. I never use the actual "Merge" option as it pollutes the history.

**Comment by majormoses** [comment]
If, you'd like to tidy up your commits I can certainly wait for you to accept this.

**Comment by absolutejam** [comment]
I didn't even know GitHub could do that...

**Comment by majormoses** [comment]
It is a somewhat new feature and one that I have been requesting for years, you can read about the different functions here: https://help.github.com/articles/about-merge-methods-on-github in my orgs I actually disable the normal merge option as merging without fast farwarding is a terrible idea.

**Comment by majormoses** [comment]
released: https://rubygems.org/gems/sensu-plugins-windows/versions/2.4.0

**Pull request merged**